core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1159,6 +1177,25 @@ class UnifiedLog(@volatile var logStartOffset: Long,
       validBytesCount, lastOffsetOfFirstBatch, Collections.emptyList[RecordError], LeaderHwChange.NONE)
   }
 
+  /**
+   * Return true if the record batch should not be appending to the log.
```

Comment:
Return true if the record batch should not be appending to the log => Return true if the record batch has a higher leader epoch than the replica?